### PR TITLE
Send email on course-creation.

### PR DESCRIPTION
**Pull request by awais786**
Send an email on new course creation.

ECOM-6801

### CODE REVIEW

**Comment by awais786** [comment]
@tasawernawaz Please review.

File: course_discovery/templates/publisher/email/course_created.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{% load i18n %}
+{% block body %}
+<!-- Message Body -->
+        <p>
+            {% blocktrans with link_start='<a href="' link_middle='">' link_end='</a>' trimmed %}
+                A course created {{ link_start }}{{ course_run_detail_page_url }}{{ link_middle }} {{ course_name }} {{ link_end }}.
+            {% endblocktrans %}
+        </p>
+
+         <!--Translators: It's closing of mail.-->
```
Comment (bderusha) [inline course_discovery/templates/publisher/email/course_created.html]: This message isn't showing up in the translators.  Please see the documentation for how to do this

https://docs.djangoproject.com/en/1.10/topics/i18n/translation/#translator-comments-in-templates

File: course_discovery/apps/publisher/tests/test_emails.py [hunk 2]
```diff
@@ -204,3 +204,70 @@ def assert_email_sent(self, object_path, subject, expected_body):
...
+        emails.send_email_for_course_creation(self.course_run.course, self.course_run)
+        # assert email sent
+        self.assert_email_sent(
+            reverse('publisher:publisher_course_run_detail', kwargs={'pk': self.course_run.id}),
+            'New Course created',
+            'A course created '
+        )
+
+    def assert_email_sent(self, object_path, subject, expected_body):
+        """ DRY method to assert sent email data"""
```
Comment (bderusha) [inline course_discovery/apps/publisher/tests/test_emails.py]: Calling a method DRY in a docstring is not really helpful documentation.  Can we update this docstring to just describe what the method does?

**Review by bderusha — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher/emails.py [hunk 3]
```diff
@@ -95,3 +95,52 @@ def send_email_for_studio_instance_created(course_run):
         email_msg.send()
     except Exception:  # pylint: disable=broad-except
         logger.exception('Failed to send email notifications for course_run [%s]', course_run.id)
+
+
+def send_email_for_course_creation(course, course_run):
+    """ Send the emails for a course creation.
+
+        Arguments:
+            course_run (CourseRun): CourseRun object
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/emails.py:105]: define `course` in arguments.

File: course_discovery/apps/publisher/emails.py [hunk 4]
```diff
@@ -95,3 +95,52 @@ def send_email_for_studio_instance_created(course_run):
...
+    """
+    try:
+        txt_template = 'publisher/email/course_created.txt'
+        html_template = 'publisher/email/course_created.html'
+
+        to_addresses = course.get_course_users_emails()
+        from_address = settings.PUBLISHER_FROM_EMAIL
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/emails.py:112]: what if `PUBLISHER_FROM_EMAIL` is not defined in settings.

File: course_discovery/apps/publisher/emails.py [hunk 5]
```diff
@@ -95,3 +95,52 @@ def send_email_for_studio_instance_created(course_run):
...
+
+        subject = _('New Course created')
+
+        email_msg = EmailMultiAlternatives(
+            subject, plain_content, from_address, to=[settings.PUBLISHER_FROM_EMAIL], bcc=to_addresses
+        )
+        email_msg.attach_alternative(html_content, 'text/html')
+        email_msg.send()
+    except Exception as e:  # pylint: disable=broad-except
+        from nose.tools import set_trace; set_trace()
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/emails.py]: remove this.

File: course_discovery/apps/publisher/emails.py [hunk 6]
```diff
@@ -95,3 +95,52 @@ def send_email_for_studio_instance_created(course_run):
...
         logger.exception('Failed to send email notifications for course_run [%s]', course_run.id)
+
+
+def send_email_for_course_creation(course, course_run):
+    """ Send the emails for a course creation.
+
+        Arguments:
+            course_run (CourseRun): CourseRun object
+    """
+    try:
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/emails.py:107]: I think we can move this `try` down where we are actually expecting exceptions.

File: course_discovery/apps/publisher/tests/test_emails.py [hunk 7]
```diff
@@ -204,3 +204,70 @@ def assert_email_sent(self, object_path, subject, expected_body):
...
+                        self.course_run.id
+                    )
+                )
+            )
+
+    def test_email_sent_successfully(self):
+        """ Verify that emails for studio instance created."""
+
+        emails.send_email_for_course_creation(self.course_run.course, self.course_run)
+        # assert email sent
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_emails.py]: Comment not needed

File: course_discovery/apps/publisher/tests/test_emails.py [hunk 8]
```diff
@@ -204,3 +204,70 @@ def assert_email_sent(self, object_path, subject, expected_body):
...
+
+
+class CourseCreatedEmailTests(TestCase):
+    """ Tests for the email functionality for new course created. """
+
+    def setUp(self):
+        super(CourseCreatedEmailTests, self).setUp()
+        self.user = UserFactory()
+
+        self.group = factories.GroupFactory()
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_emails.py]: why we are creating group here ?

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher/emails.py (hunk 6 above)
Comment (awais786) [reply course_discovery/apps/publisher/emails.py:107]: We are dealing it as a broad-exception. If any thing breaks will not send any email.

File: course_discovery/apps/publisher/emails.py (hunk 4 above)
Comment (awais786) [reply course_discovery/apps/publisher/emails.py:112]: PUBLISHER_FROM_EMAIL this will come from secure vars. And it is already added by devops.

**Comment by awais786** [comment]
@tasawernawaz @bderusha Please review this PR again.

**Comment by awais786** [comment]
````
Dear John,

Tasawer created the Computer Science course in Publisher on January 13, 2017 at 15:08:45.

Please create a Studio instance for this course.

Thanks,

The edX team
````

**Comment by awais786** [comment]
Sample email
````
Dear John,

Tasawer created the Computer Science course in Publisher on January 13, 2017 at 15:08:45.

Please create a Studio instance for this course.

Thanks,

The edX team
````

**Comment by awais786** [comment]
@bderusha @tasawernawaz Please review.

**Review by bderusha — APPROVED** [review]

File: course_discovery/apps/publisher/emails.py (hunk 4 above)
Comment (tasawernawaz) [reply course_discovery/apps/publisher/emails.py:112]: :+1:

File: course_discovery/apps/publisher/tests/test_emails.py [hunk 9]
```diff
@@ -177,7 +173,6 @@ def test_email_sent_successfully(self):
         """ Verify that emails sent successfully for studio instance created."""
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_emails.py:173]: Email will be sent on course creation or studio instance creation ?

File: course_discovery/apps/publisher/tests/test_emails.py [hunk 10]
```diff
@@ -204,3 +199,62 @@ def assert_email_sent(self, object_path, subject, expected_body):
...
+                    emails.logger.name,
+                    'ERROR',
+                    'Failed to send email notifications for course creation course run id [{}]'.format(
+                        self.course_run.id
+                    )
+                )
+            )
+
+    def test_email_sent_successfully(self):
+        """ Verify that emails for studio instance created."""
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_emails.py]: I think this email is for studio instance request which will be sent out on course creation ?

**Review by tasawernawaz — APPROVED** [review]
Do we have another sub-task to implement the final template this email? if not, please update the description of PR.

Make it sure that comments and docstrings suggests that this email is for course creation not for studio instance creation.

**Comment by awais786** [comment]
@tasawernawaz every thing is implemented I just updated the description.

File: course_discovery/apps/publisher/tests/test_emails.py (hunk 9 above)
Comment (tasawernawaz) [reply course_discovery/apps/publisher/tests/test_emails.py:173]: @awais786 Please update this comment as well. Its a small change, but will help in future.

**Pull request merged**